File: base/reverse.hs
fib n =
  if n <= 2 then
    1
  else
    fib(n-2) + fib (n-1)

reve input::String output::String = 
  if input == "" then
    output
  else
    reve (tail(input) head(input)++output)

main = print (reverse 9)

File: drill/170601/reverse.rb
def reverse(input)
    head = input[0]
    rest = input[1, input.length]
    if rest.length == 1
        return rest[0] + head
    else
        return reverse(rest) + head
    end
end

p reverse("watashinonamaehamizushima")


# time : 30s
